test/extended/openstack/loadbalancers.go

```diff
@@ -141,6 +144,59 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb] The Openstack
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(isLbMethodApplied(lbMethod, results, pods)).Should(o.BeTrue(), "%q lb-method not applied after 100 queries:\n%v\n", lbMethod, results)
 	})
+
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1997704
+	g.It("should create an UDP LoadBalancer using a pre-created FIP when an UDP LoadBalancer svc setting the LoadBalancerIP spec is created on Openshift", func() {
+
+		g.By("Create FIP to be used on the subsequent LoadBalancer Service")
+		networkIds, err := getExternalNetworkIds(networkClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		fip, err := floatingips.Create(networkClient, floatingips.CreateOpts{FloatingNetworkID: networkIds[0]}).Extract()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By(fmt.Sprintf("FIP created: %s", fip.FloatingIP))
+		defer floatingips.Delete(networkClient, fip.ID)
+
+		g.By(fmt.Sprintf("Create Openshift resources using the FIP %s on network %s", fip.FloatingIP, fip.FloatingNetworkID))
+		labels := map[string]string{"app": "udp-lb-precreatedfip-dep"}
+		depName := "udp-lb-precreatedfip-dep"
+		svcName := "udp-lb-precreatedfip-svc"
+		svcPort := int32(8022)
+		testDeployment := e2edeployment.NewDeployment(depName, 2, labels, "udp-test",
```

Comment:
Seems this deployment creation will be happening in every new test suit,  might be good to extract it to a function

Reply:
I'm addressing this on https://github.com/openshift/openstack-test/pull/59 and will adapt the rest of PRs accordingly as soon as it gets merged